### PR TITLE
Create PUT API for deletion time

### PR DESCRIPTION
**Pull request by kmdkuk**
- This is a continuation of the [previous PR discussion](https://github.com/cybozu-go/github-actions-controller/pull/35#discussion_r665074464).
- In order to unify the management of deletion time, the Annotation of deletion time has been deleted.
  - A function has been added so that the deletion time can be updated from the API.
  - Fixed to use API even in slack-agent that used annotation.

Signed-off-by: kouki <kouworld0123@gmail.com>

### CODE REVIEW

File: docs/runner-pod-api.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+* [`GET /deletion_time`](#get-deletion_time)
+* [`PUT /deletion_time`](#put-deletion_time)
+
+## `GET /deletion_time`
+
+This API indicates the pod's deletion time.
+Used by `Pod` sweeper to check the deletion time of pods.
+If the value is zero(`0001-01-01T00: 00: 00Z`), the deletion time is treated as not set.
+If the value is other than that, the deletion is judged based on whether
+or not the time has passed.
```
Comment (masa213f) [inline docs/runner-pod-api.md:13]:
```suggestion
This API returns a pod's deletion time in UTC using RFC 3339 format.

When the pod state is `initializing` or `running`, it returns the zero value of type `Time` of Go(`0001-01-01T00:00:00Z`).
When the state is `debugging` or a time is set by the [`PUT /deletion_time`](#put-deletion_time), it returns a non-zero time.

If the deletion time returned by this API has passed, the `Pod` sweeper will delete the pod.
```

File: docs/runner-pod-api.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+{
+	"deletion_time":"0001-01-01T00:00:00Z"
+}
+```
+
+## `PUT /deletion_time`
+
+This API updates the pod's deletion time.
+Used to update a pod deletion time from Extender in Slack agent.
+The time format is RFC 3339 in UTC.
```
Comment (masa213f) [inline docs/runner-pod-api.md:37]:
```suggestion
This API updates a pod's deletion time. The time format is RFC 3339 in UTC.
```

File: agent/socket.go [hunk 3]
```diff
@@ -190,3 +194,23 @@ func (s *Server) extendPod(ctx context.Context, channel, namespace, pod string,
...
+	if err != nil {
+		return err
+	}
+
+	po, err := clientset.CoreV1().Pods(namespace).Get(ctx, pod, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	runnerClient := client.NewClient()
```
Comment (zoetrope) [inline agent/socket.go:214]: Why don't you reuse the client?

Comment (kmdkuk) [reply agent/socket.go]: Thank you for your question.
I didn't have a clear reason for it.
I modify the server structure to have a runnerClient.

File: agent/socket.go [hunk 4]
```diff
@@ -190,3 +193,22 @@ func (s *Server) extendPod(ctx context.Context, channel, namespace, pod string,
...
 	return nil
 }
+
+func (s *Server) updateDeletionTime(ctx context.Context, namespace, pod string, tm time.Time) error {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
```
Comment (zoetrope) [inline agent/socket.go:203]: I think you can reuse clientset too.

Comment (kmdkuk) [reply agent/socket.go]: Thank you.
I am sorry for the same comment.
I have not been aware of the reuse of resources.

**Review by zoetrope — APPROVED** [review]

**Pull request merged**